pie-metal/src/pie_metal/_internal/pytorch_reference.py

```
@@ -0,0 +1,599 @@
+
+    # Compute cos and sin
+    cos_freqs = torch.cos(freqs)  # [num_tokens, head_size // 2]
+    sin_freqs = torch.sin(freqs)  # [num_tokens, head_size // 2]
+
+    # For non-interleaved layout (HuggingFace style), we need to concatenate freqs with itself
+    # HuggingFace does: torch.cat((freqs, freqs), dim=-1) to get [f0, f1, ..., f31, f0, f1, ..., f31]
+    # NOT repeat_interleave which would give [f0, f0, f1, f1, ...]
+    if not interleaved:
+        import os
+        if os.environ.get('DEBUG_ROPE'):
+            print(f"    [rope_reference] Using NON-INTERLEAVED mode with concatenation")
```

Ruff (0.13.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)